numpy/_core/tests/test_einsum.py

```diff
@@ -5,7 +5,8 @@
 import numpy as np
 from numpy.testing import (
     assert_, assert_equal, assert_array_equal, assert_almost_equal,
-    assert_raises, suppress_warnings, assert_raises_regex, assert_allclose
+    assert_raises, suppress_warnings, assert_raises_regex, assert_allclose,
+    assert_array_almost_equal_nulp
```

Comment:
```suggestion
    assert_raises, suppress_warnings, assert_raises_regex, assert_allclose
```
Woops.  One version preferred no buffering, and that modified the float16 results by 1ULP due to shorter inner-loops (yes, more casting due to less buffering :)).